networkoperator/v1/types_egressrouter.go

```diff
@@ -0,0 +1,274 @@
+type EgressRouterAddress struct {
+	// IP is the address to configure on the router's interface. Can be IPv4 or IPv6.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength:=1
+	IP string `json:"ip" protobuf:"bytes,1,opt,name=ip"`
+	// IP address of the next-hop gateway, if it cannot be automatically determined. Can be IPv4 or IPv6.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Gateway string `json:"gateway,omitempty" protobuf:"bytes,2,opt,name=gateway"`
+}
+
+// EgressRouterStatusConditionType is an aspect of the router's state.
+type EgressRouterStatusConditionType string
+
+const (
+	// EgressRouterAvailable indicates that the EgressRouter (the associated pod, service, NAD), is functional and available in the cluster.
+	EgressRouterAvailable EgressRouterStatusConditionType = "Available"
+
+	// EgressRouterProgressing indicates that the router is actively rolling out new code,
+	// propagating config changes, or otherwise moving from one steady state to
+	// another.
+	EgressRouterProgressing EgressRouterStatusConditionType = "Progressing"
+
+	// EgressRouterDegraded indicates that the router's current state does not match its
+	// desired state over a period of time resulting in a lower quality of service.
+	EgressRouterDegraded EgressRouterStatusConditionType = "Degraded"
+)
+
+// ConditionStatus defines the status of each of EgressRouterStatusConditionType.
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a resource is in the condition.
+// "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes
+// can't decide if a resource is in the condition or not. In the future, we could add other
+// intermediate conditions, e.g. ConditionDegraded.
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+// EgressRouterStatusCondition represents the state of the egress router's
+// managed and monitored components.
+// +k8s:deepcopy-gen=true
+type EgressRouterStatusCondition struct {
+	// Type specifies the aspect reported by this condition; one of Available, Progressing, Degraded
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="Available";"Progressing";"Degraded"
+	// +required
+	Type EgressRouterStatusConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=EgressRouterStatusConditionType"`
+
+	// Status of the condition, one of True, False, Unknown.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="True";"False";"Unknown"
+	// +required
+	Status ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=ConditionStatus"`
+
+	// LastTransitionTime is the time of the last update to the current status property.
+	// +kubebuilder:validation:Required
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime" protobuf:"bytes,3,opt,name=lastTransitionTime"`
```

Comment:
you probably want to make this nullable because `metav1.Time` renders as `null` when not set.

Reply:
oh yes I will make it nullable, nice catch!